input/resources/au-core-medicationrequest.xml

```diff
@@ -248,7 +248,7 @@
         <key value="au-core-medreq-03"/>
         <severity value="error"/>
         <human value="Date shall be precise to the day or, if not available, the Data Absent Reason extension shall be present"/>
-        <expression value="(toString().length() &gt;= 8) xor extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists()"/>
+        <expression value="(value.exists() and value.toString().length() &gt;= 10 and extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists().not()) or (value.empty() and extension('http://hl7.org/fhir/StructureDefinition/data-absent-reason').exists())"/>
         <xpath value="(f:matches(f:effectiveDateTime, /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z)) and not(exists(f:extension[@url='http://hl7.org/fhir/StructureDefinition/data-absent-reason']))) or (not(f:matches(f:effectiveDateTime, /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))) and exists(f:extension[@url='http://hl7.org/fhir/StructureDefinition/data-absent-reason']))"/>
```

Comment:
please update xpath to match the updated expression.

Reply:
I am not convinced this is the solution. Could not get it to work with FHIRPath Lab either. 
I've raised a question in Zulip: [MedicationRequest.authoredOn allow precise date or DAR ext](https://chat.fhir.org/#narrow/channel/179266-fhirpath/topic/MedicationRequest.2EauthoredOn.20allow.20precise.20date.20or.20DAR.20ext); @JayMurdoch, could you please follow up if there is no response?